designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/FullScreenImage.kt

```diff
@@ -0,0 +1,167 @@
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.launch
+import org.hisp.dhis.mobile.ui.designsystem.component.internal.image.ZoomableImage
+import org.hisp.dhis.mobile.ui.designsystem.theme.Spacing
+import org.hisp.dhis.mobile.ui.designsystem.theme.SurfaceColor
+
+/**
+ * DHIS2 Full Screen Image. Wraps DHIS2 Internal [ZoomableImage].
+ * @param painter controls the image painter to be drawn
+ * @param title controls the text to be shown for the title
+ * @param modifier allows a modifier to be passed externally
+ * @param onDismiss is a callback to dismiss the full screen image component
+ * @param onDownloadButtonClick is a callback to notify when the download button is clicked.
+ * @param onShareButtonClick is a callback to notify when the share button is clicked.
+ */
+@Composable
+fun FullScreenImage(
+    painter: Painter,
+    title: String,
+    modifier: Modifier = Modifier,
+    onDismiss: () -> Unit,
+    onDownloadButtonClick: () -> Unit,
+    onShareButtonClick: () -> Unit,
+) {
+    val animatedScale = remember { Animatable(0f) }
+    val animatedColor = remember { Animatable(Color.Transparent) }
+
+    LaunchedEffect(animatedScale, animatedColor) {
+        launch {
+            awaitAll(
+                async {
+                    animatedScale.animateTo(
+                        1f,
+                        animationSpec = tween(500),
+                    )
+                },
+                async {
+                    animatedColor.animateTo(
+                        Color.Black,
+                        animationSpec = tween(500),
+                    )
+                },
+
+            )
+        }
+    }
+
+    Dialog(
+        properties = DialogProperties(
+            usePlatformDefaultWidth = false,
+            dismissOnClickOutside = false,
+        ),
+        onDismissRequest = onDismiss,
+    ) {
+        Box(
+            modifier
+                .background(color = animatedColor.value),
+            contentAlignment = Alignment.Center,
+        ) {
+            ZoomableImage(
+                painter = painter,
+                modifier = Modifier
+                    .testTag("FULL_SCREEN_IMAGE")
+                    .fillMaxSize()
+                    .graphicsLayer {
+                        scaleX = animatedScale.value
+                        scaleY = animatedScale.value
+                    },
+            )
+            AppBar(
+                title = title,
+                modifier = Modifier
+                    .align(Alignment.TopStart),
+                onBack = onDismiss,
+                onShare = onShareButtonClick,
+                onDownload = onDownloadButtonClick,
+            )
+        }
+    }
+}
+
+@Composable
+private fun AppBar(
```

Comment:
Why haven't used the `TopAppBar` from Material package? Is there anything that is not customisable in there?

Reply:
@msasikanth Done. AppBar has been replaced by TopAppBar